addons/im_livechat/static/tests/embed/chat_window.test.js

```diff
@@ -114,3 +114,27 @@ test("avatar url contains access token for non-internal users", async () => {
         }/avatar_128?access_token=${guest.id}&unique=${deserializeDateTime(guest.write_date).ts}"]`
     );
 });
+
+test("Open confirm dialog on Esc, cancel it on Esc/ confirm it on Enter", async () => {
+    await startServer();
+    await loadDefaultEmbedConfig();
+    onRpcBefore((route, args) => {
+        if (route === "/im_livechat/visitor_leave_session") {
+            step(route);
+        }
+    });
+    await mountWithCleanup(LivechatButton);
+    await click(".o-livechat-LivechatButton");
+    await contains(".o-mail-ChatWindow");
+    await insertText(".o-mail-Composer-input", "Hello");
+    await triggerHotkey("Enter");
+    await contains(".o-mail-Message", { text: "Hello" });
+    await triggerHotkey("Escape");
+    await contains(".o-livechat-CloseConfirmation");
+    await triggerHotkey("Escape");
+    await contains(".o-livechat-CloseConfirmation", { count: 0 });
+    await triggerHotkey("Escape");
+    await contains(".o-livechat-CloseConfirmation");
+    await triggerHotkey("Enter");
+    await assertSteps(["/im_livechat/visitor_leave_session"]);
```

Comment:
Assert the chat window is no longer there, there's the notification to tell the conversation has left. The rpc assert step is nice, but that's invisible to end-user so that's more a technical assertion the rpc was made (which doesn't necessarily mean the leave really happens... to play devil's avocate, we trust the server code that this route will do the actual leave).

Reply:
The chat window is still there in this step. I added a text assert on the chat window content.

Reply:
Oh yes my bad, I was thinking from operator's PoV. Indeed for visitors this is still open with feedback panel.